enctests/testframework/generatetests.py

```diff
@@ -0,0 +1,149 @@
+                    info = ""
+                else:
+                    incommand = True
+                    lastcommandstart = linenumber
+                continue
+            if incomment:
+                info = info + line
+            if incommand:
+                command = command + line
+
+    # Now we process the scanned data to create the config file.
+    tests = []
+
+    for testcount, test in enumerate(allinfo):
+        testconfigs = {}
+
+        testname = test['config'].get("name", "test-%d" % (testcount + 1))
+        outputfile = test['command'].split(" ")[-1]
+        outputfileext = outputfile.split(".")[-1]
+        #print("START:", test['command'])
+        template = re.sub("\s\S+$", ' {encoding_args} -y "{outfile}"', test['command'])
+        template = re.sub("\s-i\s\S+\s", ' {input_args} -i "{source}" ', template)
+        #print("Pre vframes:", template)
+        template = re.sub("\s-vframes\s\S+\s", ' -vframes {duration} ', template)
+        #print("Post vframes:", template)
+
+        # input arg flags.
+        for arg in ['-r', '-start_number']:
+            template = re.sub("\s%s\s\S+\s" % arg, ' ', template) 
+        
+        wedge = {}
+
+        # Main args.
+        for arg in ["-vf", '-color_range', '-colorspace', '-color_primaries', '-profile:v', '-compression_level', '-pred', '-color_trc', '-pix_fmt', '-preset', '-c:v', '-sws_flags']:
+            match = re.search("\s%s\s(\S+)\s" % arg, template)
+            if match:
+                wedge[arg] = match.group(1)
+                template = template.replace(match.group(0), ' ')
+        
+        #print("Wedge:", wedge)
+
+
+        print("Name:", testname)
+        print("Template:", template)
+        if "sources" in test['config']:
+            for i in range(0, len(test['config']['sources'])):
+                test['config']['sources'][i] = os.path.abspath(os.path.join(args.root, test['config']['sources'][i]))
+        testconfigs[testname] = test['config']
+        testconfigs[testname]['encoding_template'] = template
+        testconfigs[testname]['suffix'] = "." + outputfileext
+        testconfigs[testname]['description'] = "Test from %s line %d" % (test['file'], test['line'])
+        testconfigs[testname]['name'] = testname
+        testconfigs[testname]['app'] = 'ffmpeg'
+        if 'wedges' in testconfigs[testname]:
+            # We allow additional wedges to be added in the docs config file.
+            print("Already got wedge:", testconfigs[testname]['wedges'])
+            testconfigs[testname]['wedges']['wedge0'] = wedge
+        else:
+            testconfigs[testname]['wedges'] = {'wedge0': wedge}
+        tests.append(testconfigs)
+
+    reports = {'reports': {'name': 'doc-tests',
+                            'directory': 'docs-encode',
+                            'templatefile': 'doctests.html.jinja'}}
+    tests.append(reports)
+
+    f = open(args.output_config_file, "w")
+    yaml.dump_all(tests, f)
```

Comment:
Same comment regarding closing files. The recommended way to open/close files is:

```
with open(args.output_config_file, "w") as f:
    yaml.dump_all(tests, f)
```

Reply:
Should be addressed.